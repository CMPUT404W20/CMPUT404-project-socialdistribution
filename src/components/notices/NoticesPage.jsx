import React, { Component } from "react";
import "../../styles/friends/FriendsPage.scss";
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import NotificationsNoneOutlinedIcon from "@material-ui/icons/NotificationsNoneOutlined";
import NavigationBar from "../NavigationBar";
import NoticeItem from "./NoticeItem";

class NoticesPage extends Component {
  constructor(props) {
    super(props);
    this.props = props;
    this.state = {
      noticesList: [{ id: "001", name: "Username1", type: "Local" }, { id: "002", name: "Username2", type: "Local" }, { id: "003", name: "Username3", type: "Local" }, { id: "004", name: "Username4", type: "Local" }, { id: "005", name: "Username5", type: "Local" }],
    };
  }

  handleRemoveNotice(id) {
    const { noticesList } = this.state;
    const filteredList = noticesList.filter((item) => item.id !== id);
    this.setState({ noticesList: filteredList });
  }

  handleAccept(id) {
    // accept the request
    this.handleRemoveNotice(id);
  }

  renderNoticeItems = () => {
    const { noticesList } = this.state;
    return (
      <Row className="item-list">
        {noticesList.map((item) => (
          <NoticeItem
            key={item.id}
            username={item.name}
            id={item.id}
            type={item.type}
            handleAccept={(id) => this.handleAccept(id)}
            handleDecline={(id) => this.handleRemoveNotice(id)}
          />
        ))}
      </Row>
    );
  }

  render() {
    const { noticesList } = this.state;
    return (
      <Container fluid className="page-wrapper">
        <Row>
          <Col md={12}>
            <NavigationBar selected="Notifications" />
          </Col>
        </Row>
        <Row>
          <Col md={2} />
          <Col md={8}>
            <div className="content-wrapper">
              <div className="title">
                <NotificationsNoneOutlinedIcon className="icon" />
                <p>Notifications</p>
                <p>
                  (
                  {noticesList.length}
                  )
                </p>
              </div>
              {this.renderNoticeItems()}
            </div>
          </Col>
        </Row>
      </Container>
    );
  }
}
export default NoticesPage;
