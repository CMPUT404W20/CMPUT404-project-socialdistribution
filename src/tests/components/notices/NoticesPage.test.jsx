import React from "react";
import { shallow } from "enzyme";
import NoticesPage from "../../../components/notices/NoticesPage";

describe("Notices Page Component", () => {
  it("should render correctly", () => {
    const component = shallow(<NoticesPage />);
    expect(component).toMatchSnapshot();
  });
});
