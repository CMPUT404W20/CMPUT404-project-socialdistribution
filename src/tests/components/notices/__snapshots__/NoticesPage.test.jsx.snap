// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notices Page Component should render correctly 1`] = `
<Container
  className="page-wrapper"
  fluid={true}
>
  <Row
    noGutters={false}
  >
    <Col
      md={12}
    >
      <NavigationBar />
    </Col>
  </Row>
  <Row
    noGutters={false}
  >
    <Col
      md={2}
    />
    <Col
      md={8}
    >
      <div
        className="content-wrapper"
      >
        <div
          className="title"
        >
          <NotificationsNoneOutlinedIcon
            className="icon"
          />
          <p>
            Notifications
          </p>
          <p>
            (
            5
            )
          </p>
        </div>
        <Row
          className="item-list"
          noGutters={false}
        >
          <NoticeItem
            handleAccept={[Function]}
            handleDecline={[Function]}
            key="001"
            type="Local"
            userID="001"
            username="Username1"
          />
          <NoticeItem
            handleAccept={[Function]}
            handleDecline={[Function]}
            key="002"
            type="Local"
            userID="002"
            username="Username2"
          />
          <NoticeItem
            handleAccept={[Function]}
            handleDecline={[Function]}
            key="003"
            type="Local"
            userID="003"
            username="Username3"
          />
          <NoticeItem
            handleAccept={[Function]}
            handleDecline={[Function]}
            key="004"
            type="Local"
            userID="004"
            username="Username4"
          />
          <NoticeItem
            handleAccept={[Function]}
            handleDecline={[Function]}
            key="005"
            type="Local"
            userID="005"
            username="Username5"
          />
        </Row>
      </div>
    </Col>
  </Row>
</Container>
`;
